locations/spiders/sunglass_hut_2.py

```diff
@@ -0,0 +1,45 @@
+from scrapy import Spider
+
+from locations.dict_parser import DictParser
+from locations.hours import DAYS_FULL, OpeningHours
+
+
+class SunglassHut2Spider(Spider):
+    name = "sunglass_hut_2"
+    allowed_domains = [
+        "mena.sunglasshut.com",
+        "nl.sunglasshut.com",
+        "pt.sunglasshut.com",
+        "tr.sunglasshut.com",
+        "hk.sunglasshut.com",
+    ]
+
+    item_attributes = {"brand": "Sunglass Hut", "brand_wikidata": "Q136311"}
+
+    start_urls = [
+        "https://mena.sunglasshut.com/api/content/render/false/limit/9999/type/json/query/+contentType:SghStoreLocator%20+languageId:11%20+deleted:false%20+working:true/orderby/modDate%20desc",
+        "https://nl.sunglasshut.com/api/content/render/false/limit/9999/type/json/query/+contentType:SghStoreLocator%20+languageId:3%20+deleted:false%20+working:true/orderby/modDate%20desc",
+        "https://pt.sunglasshut.com/api/content/render/false/limit/9999/type/json/query/+contentType:SghStoreLocator%20+languageId:4%20+deleted:false%20+working:true/orderby/modDate%20desc",
+        "https://tr.sunglasshut.com/api/content/render/false/limit/9999/type/json/query/+contentType:SghStoreLocator%20+languageId:4%20+deleted:false%20+working:true/orderby/modDate%20desc",
+        "https://hk.sunglasshut.com/api/content/render/false/limit/9999/type/json/query/+contentType:SghStoreLocator%20+languageId:8%20+deleted:false%20+working:true/orderby/modDate%20desc",
+    ]
+
+    def parse(self, response):
```

Comment:
Oops I missed something in my review, and it's causing 1 result to return because there is a dupefilter that thinks all locations are the same without a distinguishing ID.

We need to set `item["ref"]` manually in this case, and there is no ID field. Instead I suggest that the `seoUrl` field be used, similar to `item["website"] = response.url.split(...) + data["seoUrl"]` and then `item["ref"] = item["website"]`.

Reply:
Oh, I didn't see that. The following should be better, as the websites would also be legit.
I did not add the website previously, as these are not exactly _websites_, they just provide info about the specific store.

```suggestion
    def parse(self, response):
      region=response.url.split("https://", 1)[1].split(".", 1)[0].lower()
      item["ref"]=f"{region}-{data['seoUrl']}"
      
      REGION_TO_LANGUAGE={"mena":"en" , "nl":"nl" ,...} 
      item["website"]="https://{}.sunglasshut.com/{}/store-locator/{}".format(region,REGION_TO_LANGUAGE[region],data['seoUrl'])
```

the websites would be like this:
https://mena.sunglasshut.com/en/store-locator/riyadh-panorama
https://nl.sunglasshut.com/nl/store-locator/lelystad-lelystad

Reply:
Those websites are fine to use. Looks good!